src/xsarsea/cross_spectra_core.py

```diff
@@ -0,0 +1,307 @@
+"""
+author Antoine Grouazel:
+original code provided by Frederic Nouguier the 29 March 2022
+"""
+import numpy as np
+import xarray as xr
+import scipy
+import matplotlib.pyplot as plt
+import xrft
+import logging
+import xsar
+import os
+
+def get_imagette_indice(onetiff,wv_slc_meta):
+    good_indice = None
+    # find the indice of the tiff
+    imagette_number = os.path.basename(onetiff).split('-')[-1].replace('.tiff','')
+    logging.info('imagette_number : %s',imagette_number)
+    for ddi,ddname in enumerate(wv_slc_meta.subdatasets) :
+        if 'WV_'+imagette_number in ddname :
+            logging.info("matching ddname : %s",ddname)
+            good_indice = ddi
+    logging.info('indice in meta sentinel1 gdal driver is : %s',good_indice)
+    return good_indice
+
+def read_slc(onetiff,slice_subdomain=None,resolution=None,resampling=None):
+    """
+
+    :param one_wv: SAFE (str) full path without the last /
+    :param slice_subdomain : slice objet to define the sub part of imagette to perform cross spectra
+    :param resolution: dict for instance for 200m {'atrack' : int(np.round(200 / sar_meta.pixel_atrack_m)), 'xtrack': int(np.round(200 / sar_meta.pixel_xtrack_m))}
+    :param rasterio.enums.Resampling.rms for instance
+    :return:
+    """
+    logging.info('tiff: %s',onetiff)
+    logging.info('resampling : %s',resampling)
+    safepath = os.path.dirname(os.path.dirname(onetiff))
+    logging.info('safepath: %s',safepath)
+    wv_slc_meta = xsar.Sentinel1Meta(safepath)
+    logging.debug('wv_slc_meta %s',wv_slc_meta)
+    good_indice = get_imagette_indice(onetiff,wv_slc_meta)
+    if resolution is not None:
+        if resolution['atrack']==1 and resolution['xtrack']==1:
+            #June 2021, a patch because currently resolution 1:1 for image and rasterio returns an error
+            slc = xsar.Sentine1Dataset(wv_slc_meta.subdatasets[good_indice],resolution=None,resampling=None)
+        else:
+            slc = xsar.Sentine1Dataset(wv_slc_meta.subdatasets[good_indice],resolution=resolution,resampling=resampling)
+    else:
+        slc = xsar.Sentine1Dataset(wv_slc_meta.subdatasets[good_indice])
+    logging.info('max xtrack val : %s',slc['xtrack'].values[-1])
+    slc = slc.rename({'atrack' : 'azimuth','xtrack' : 'range'})
+    azimuthSpacing,rangeSpacing = slc.s1meta.image['slant_pixel_spacing']
```

Comment:
Do you think that putting those attributes in the xarray.Dataset object is possible ? If so, we don't need to pass a Sentinel1Dataset object, but just an xarray object

Reply:
No. Just access them from xarray.Dataset attribute.
Something like:
```python
azimuthSpacing = s1ds.attrs['azimuthSpacing']
rangeSpacing = s1ds.attrs['azimuthSpacing']
```

So no needs to use `s1meta` object: just use attributes from xarray Dataset (Note that you will need to make those attributes available in xsar)

Reply:
I opened an issue https://github.com/umr-lops/xsar/issues/100 to track the changes needed.